packages/jupyterlab-chat-extension/package.json

```
@@ -48,8 +48,8 @@
   },
   "dependencies": {
     "@jupyter-notebook/application": "^7.2.0",
-    "@jupyter/collaborative-drive": "^3.0.0",
-    "@jupyter/ydoc": "^2.0.0 || ^3.0.0",
+    "@jupyter/collaborative-drive": "^4.0.2",
+    "@jupyter/ydoc": "^2.1.3 || ^3.0.0",
```

For packages that have frontend & backend components, I think the NPM and PyPI version ranges should match. `pyproject.toml` requires `jupyter_ydoc>=3.0.0,<4.0.0`, so this version range may need to be updated:

```suggestion
    "@jupyter/ydoc": "^3.0.0",
```

`@jupyter/ydoc^3.0.0` seems compatible with all JL4 versions too, so this should be safe: https://github.com/jupyter-server/jupyter_ydoc/blob/0481b0782d16225fb8bc3f87bab3c993b64c6586/javascript/package.json#L43-L50